src/pytorch_lightning/loggers/neptune.py

```diff
@@ -618,18 +619,18 @@ def _dict_paths(cls, d: dict, path_in_build: str = None) -> Generator:
     @property
     def name(self) -> str:
         """Return the experiment name or 'offline-name' when exp is run in offline mode."""
-        return self._run_name
+        return self._run_name  # type: ignore[return-value]
 
     @property
     def version(self) -> str:
         """Return the experiment version.
 
         It's Neptune Run's short_id
         """
-        return self._run_short_id
+        return self._run_short_id  # type: ignore[return-value]
```

Comment:
`_run_sort_id` is annotated with Optional[None] in \_\_init_\_\, and is then set to a string in `._retrieve_run_data`.  mypy is incorrectly evaluating `_run_sort_id` as Optional[str]

Reply:
This should already be `str` by the time it's used, shouldn't it? If so, I'm voting for `assert` instead of `# type: ignore`

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13692/commits/575f150fdb8cd7406d3a04f43a35be1fb1e63618

Reply:
I took the liberty to update the code a bit to invoke `self.run` in order to fetch `self._run_name` and `self._run_short_id` to make the tests happy again